blocks.go

```diff
@@ -2,47 +2,43 @@ package main
 
 import (
 	"context"
-	"fmt"
 	"notes-service/models"
 	notespb "notes-service/protorepo/noted/notes/v1"
+	"notes-service/validators"
 	"strconv"
 
-	"github.com/google/uuid"
 	"go.uber.org/zap"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
-	"google.golang.org/protobuf/types/known/emptypb"
 )
 
-type blocksService struct {
-	notespb.UnimplementedNotesAPIServer
-
-	logger *zap.Logger
-	repo   models.BlocksRepository
-}
-
-var _ notespb.NotesAPIServer = &notesService{}
-
-func (srv *blocksService) InsertBlock(ctx context.Context, in *notespb.InsertBlockRequest) (*notespb.InsertBlockResponse, error) {
-	_, err := uuid.Parse(strconv.Itoa(int(in.NoteId)))
+func (srv *notesService) InsertBlock(ctx context.Context, in *notespb.InsertBlockRequest) (*notespb.InsertBlockResponse, error) {
+	token, err := srv.authenticate(ctx)
 	if err != nil {
-		srv.logger.Error("invalid uuid :", zap.Error(err))
-		return nil, status.Errorf(codes.InvalidArgument, "could not insert block")
+		return nil, status.Error(codes.Unauthenticated, err.Error())
 	}
-
-	if in.Block.Data == nil || in.Index < 1 || in.Block.Type < 1 {
-		srv.logger.Error("invalid arguments")
-		return nil, status.Errorf(codes.InvalidArgument, "could not insert block")
+	err = validators.ValidateInsertBlockRequest(in)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+	//check si la note appartient a celui qui veut la modifier
+	note, err := srv.repoNote.Get(ctx, strconv.Itoa(int(in.NoteId)))
+	if err != nil {
+		srv.logger.Error("Note not found in database", zap.Error(err))
+		return nil, status.Error(codes.NotFound, "could not update note")
```

Comment:
Check out the `accounts-service`'s `statusFromModelError` function. It's a little utility function that converts an error from a repository into a gRPC error. It's useful because it differentiates between not found errors and the rest. So the server can return the appropriate status code: `HTTP 404` if it's truly not found (user error), or `HTTP 500` if there was another failure (such as failing to connect to db or else, which is a server error). Error codes are important because the frontend might retry a query that ends in a `HTTP 500` code but will not retry a query that ended in a `HTTP 404`. Besides it's misleading if we're trying to debug. You don't have to fix it now, I'll take care of fixing it next sprint, just so you know.